esp/esp/utils/latex.py

```diff
@@ -170,8 +170,17 @@ def _gen_latex(texcode, stdout, stderr, type='pdf'):
     elif retcode:
         # Otherwise, if there was an error, we want to exit now since things
         # didn't work.
-        # TODO(benkraft): Return part of the log file here.
-        raise ESPError("LaTeX failed; try looking at the log file.")
+        # TODO(benkraft): Try to extract the actual error out.
+        raise ESPError('LaTeX failed; try looking at the log file.  Here are '
+                       'the last 1000 characters of the log: %s'
+                       % tex_log[-1000:])
```

Comment:
![screenshot from 2016-12-11 11-31-56](https://cloud.githubusercontent.com/assets/111238/21082689/75d9e060-bf95-11e6-9c95-bbae9356e1fc.png)

I think I'm gonna call this good enough for now -- 1000 characters isn't that much spew, it's way better than it used to be, and short of a custom error page I don't see a great way to do better since I don't think we can do any real formatting from here.